src/sentry/notifications/notificationcontroller.py

```diff
@@ -169,224 +171,223 @@ def _get_layered_setting_options(
                         continue
-                    for provider in ExternalProviderEnum:
-                        most_specific_setting = None
-                        for setting in scoped_settings:
-                            user_or_team_settings = []
-                            if recipient_is_user(recipient):
-                                user_or_team_settings = self._filter_providers(
-                                    settings=setting,
-                                    user_id=recipient.id,
-                                    type=type.value,
-                                    provider=provider.value,
-                                )
-                            elif recipient_is_team(recipient):
-                                user_or_team_settings = self._filter_providers(
-                                    settings=setting, team_id=recipient.id, type=type.value
-                                )
-
-                            if len(user_or_team_settings) > 0:
-                                most_specific_setting = NotificationSettingsOptionEnum(
-                                    user_or_team_settings[0].value
-                                )
-                                break
-
-                        if most_specific_setting is None:
-                            most_specific_setting = get_default_for_provider(type, provider)
-
-                        layered_setting_providers[recipient][scope][type].update(
-                            {provider: most_specific_setting}
-                        )
-
-        return layered_setting_providers
 
-    def get_notification_recipients(
+                # sort the settings by scope type, with the most specific scope last so we override with the most specific value
+                most_specific_recipient_providers[NotificationSettingEnum(setting.type)][
+                    ExternalProviderEnum(setting.provider)
+                ] = NotificationSettingsOptionEnum(setting.value)
+
+            # if we have no settings for this user, use the defaults
+            for type in NotificationSettingEnum:
+                for provider_str in PERSONAL_NOTIFICATION_PROVIDERS:
+                    provider = ExternalProviderEnum(provider_str)
+                    if provider not in most_specific_recipient_providers[type]:
+                        # TODO: fix so team defaults don't have different logic
+                        if recipient_is_team(recipient):
+                            most_specific_recipient_providers[type][
+                                provider
+                            ] = NotificationSettingsOptionEnum.NEVER
+                        else:
+                            most_specific_recipient_providers[type][
+                                provider
+                            ] = get_default_for_provider(type, provider)
+
+        return most_specific_setting_providers
+
+    def get_combined_settings(
```

Comment:
`get_combined_settings` doesn't quite communicate that we're filtering out settings that are off.